proto_parser/demo/demo_sample_generator_main.cc

```diff
@@ -0,0 +1,107 @@
+/*
+*   Copyright 2020 Google LLC
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       https://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*/
+
+#include <filesystem>
+#include <fstream>
+
+#include "../src/proto_value.h"
+#include "../src/proto_parser.h"
+#include "proto/event_predicate.pb.h"
+
+using ::wireless_android_play_analytics::ProtoValue;
+using ::wireless_android_play_analytics::MessageValue;
+using ::wireless_android_play_analytics::PrimitiveValue;
+using ::wireless_android_play_analytics::EventPredicate;
+using ::wireless_android_play_analytics::ProtoParser;
+
+namespace {
+
+std::string Generate(MessageValue* message) {
```

Comment:
MessageValue* ->const MessageValue&

Reply:
Here we are in fact modifying the MessageValue.